bitcoin/src/util/hash.rs

```diff
@@ -123,4 +162,75 @@ mod tests {
         let from_array = bitcoin_merkle_root_inline(&mut hashes_array);
         assert_eq!(from_iter, from_array);
     }
+
+    #[test]
+    fn test_merkle_root_from_path() {
+        let coinbase_hash = [
```

Comment:
could you add a comment with reference to the tx/block if they are available on mainnet or testnet?

Reply:
they are not, what I could do is to add a function to calculate path from ids and check that
```rust
let path = path_from_ids(real_block.ids, index);
let expected = bitcoin_merkle_root(real_block.merkle_root);
let root = bitcoin_merkle_root_from_path(path);
assert_eq!(expected,root)
```

What about that?